documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_moderation_metrics.md

```diff
@@ -0,0 +1,327 @@
+
+This implementation plan deals with the development of various metrics, and
+appropriate channels to surface them to the team, which will enable us to
+monitor the performance and effectiveness of the content moderation pipeline.
+
+These metrics can be helpful for visualising both the accuracy and volume of
+incoming reports, and the speed and volume of our response to these reports.
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+The following views will be created as a realisation of this plan.
+
+- Real-time dashboard in AWS CloudWatch
+- Periodic/on-demand reports in Django Admin
+
+Read on for more information about these views and the various metrics displayed
+by each.
+
+## Step-by-step plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+```{mermaid}
+flowchart TD
+  subgraph b[Deferred metrics]
+    4[Step 4: Update models] --- 5[Step 5: Implement deferred metrics] --- 6[Step 6: Surface in Django Admin]
+  end
+  subgraph a[Realtime metrics]
+    2[Step 2: Implement real-time metrics] --- 3[Step 3: Surface in AWS CloudWatch]
+  end
+  1[Step 1: Identify and categorise metrics] --- 2 & 4
+```
+
+In these steps, step 1 should be done sufficiently by this plan itself.
+
+Steps 2 and 4 need step 1 to be completed but can be done independently of each
+other. Steps 3 and 5 depends on 2 and 4 respectively. Hence, there can be two
+parallel tracks of work, one for the real time metrics and one for the deferred
+metrics.
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+```{caution}
+This document shows code-samples but intended as a guide and not meant to be
+used directly!
+```
+
+### Identify and categorise metrics
+
+The metrics we aim to measure can be classified along two axes:
+
+- Whether they measure user-submitted reports or moderator-generated responses
+- Whether they are real-time or deferred
+
+The following metrics can be considered to begin with (we can always add more in
+the future):
+
+- Metrics about reports
+  - number of reports (time-series, real-time)
+    <!-- TODO: - per media type (time-series, real-time) -->
+    - per violation (copyright, sensitive, other)
+  - most reported creator/source/provider (deferred)
+  - accuracy of reports (deferred)
```

Comment:
In this IP, I've decided that we'll count those reports as accurate because the `decision_id` associated with them corresponds to a confirmation decision. A later decision with `reversed_mark_sensitive` or `reversed_deindex` will not necessarily map to the original reports that got the media item marked sensitive or deindexed so there is no way to count them as inaccurate.